matchbox/provider_test.go

```diff
@@ -0,0 +1,251 @@
+package matchbox
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net"
+	"testing"
+
+	"github.com/coreos/matchbox/matchbox/rpc"
+	"github.com/coreos/matchbox/matchbox/server"
+	"github.com/coreos/matchbox/matchbox/storage"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+var providers = map[string]terraform.ResourceProvider{
+	"matchbox": Provider(),
+}
+
+func TestProvider(t *testing.T) {
+	if err := Provider().(*schema.Provider).InternalValidate(); err != nil {
+		t.Fatalf("err: %s", err)
+	}
+}
+
+type MatchboxFixtureServer struct {
+	Store    storage.Store
+	Listener net.Listener
```

Comment:
```
type MatchboxFixtureServer struct {
    TLSInfo TLSInfo
    Store storage.Store
}
```
Possibly also *grpc.Server so we can hold onto the server and close it properly at the end of tests.